Aug 4 2021/Ch1ResnetV3.py

```diff
@@ -1,245 +1,225 @@
+"""
+hyperparameter = dict(  # hyper parameters dictionary
+    learning_rate=1e-3, # learning rate
+    batch_size=16, # patch size
+    num_epochs=1, # number of epochs
+    num_workers=1, # number of workers
 )
+# {dict:4} dict with 4 elements {'learning_rate':0.001,'batch_size':16,'num_epochs':1,'num_workers':1}
+""" Variables and paths to data """
 
-"""
-Variables 
-"""
-
+"""Paths to files where the data is located  """
 csv = '/home/zachary/Desktop/DeepLearning/PreProcessing/CAKI2/CAKI2.csv'   # Path to csv file with labels and file names
 Ch1 = '/home/zachary/Desktop/DeepLearning/Dataset/CAKI2/All/Ch1'   # Path to Ch1 images.
-train = '/home/zachary/Desktop/DeepLearning/PreProcessing/CAKI2/train/train.csv' # Path to train csv file
+train = '/home/zachary/Desktop/DeepLearning/PreProcessing/CAKI2/train/train.csv' # Path to training csv file
 validation = '/home/zachary/Desktop/DeepLearning/PreProcessing/CAKI2/test/test.csv' # path to validation csv file
-loss_list = [] # loss list
-validation_loss_list = [] # validation lost list
-# Transforms
-transform = transforms.Compose([
-            transforms.ToPILImage(),
-            transforms.Resize((256, 256)),
-            transforms.CenterCrop(224),
-            transforms.ToTensor()
-        ])
-train_set = CellDataset(csv_file=train, root_dir=Ch1, transform=transform)
-validation_set = CellDataset(csv_file=validation, root_dir=Ch1, transform=transform)
-"""
-Step 5 Mode Pipeline 
-"""
-
-
-"""
-Step 5 Mode Pipeline 
-"""
-def model_pipeline(hyperparameters):
-    # Tell wandb to get started
-    with wandb.init(project='test', config=hyperparameters):
-        # acess all HPs through wandb.config, so logging matches execution
-        config = wandb.config
-        # print(config) # Prints the config dict defined above
-        # make the model,data and optimization problem
-        model, train_loader, validation_loader, criterion, optimizer=make(config)
-        # and use them to train the model
-        wandb.watch(model, criterion, log='all', log_freq=10)
-        for epoch in range(config.num_epochs):
-
-            train(model, train_loader, criterion, optimizer, config)
-            # and test its final performance
-            validation(model, validation_loader, criterion)
-    return model
-
-
-"""
-Step 6 Make the data, model , loss and optimizer 
-"""
 
-
-def make(config):
-    # Make the data
-    train_loader = DataLoader(train_set, batch_size=config.batch_size, shuffle=True, num_workers=config.num_workers)
-    validation_loader = DataLoader(validation_set, batch_size=config.batch_size, shuffle=False, num_workers=config.num_workers)
-
-    # Make the model
-    model = torchvision.models.resnet18(pretrained=True)
-    model.conv1 = nn.Conv2d(1, 64, (7, 2), padding=3, bias=False)
-    model.fc = nn.Linear(in_features=512, out_features=1)
-    model.to(device)
+"""Variables"""
+loss_list = [] # Empty list used to store loss during training list
```

Comment:
I would move the definition of loss_list to the begining of each training epoch, aka line 136. You want to define an empty list at the beginning of each epoch, append the loss to the list each step, and then average at the end of each epoch. E.g., if you train for 100 epochs, you should define loss_list 100 times and log it to wandb only 100 times.